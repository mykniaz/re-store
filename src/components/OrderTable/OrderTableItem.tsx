import * as React from 'react';
import { IOrderTableItem } from './OrderTable';

interface IProps {
  col: IOrderTableItem;
  colIndex: number;
  onIncrease: (id: number) => void;
  onDecrease: (id: number) => void;
  onDelete: (id: number) => void;
}

const OrderTableItem: React.SFC<IProps> = ({ col, colIndex, onDecrease, onIncrease, onDelete }) => {
  return (
    <tr>
      <th scope="row">{colIndex + 1}</th>
      <td>{col.title}</td>
      <td>{col.count}</td>
      <td>{col.total}$</td>
      <td>
        <div className="btn-group">
          <button
            className="btn btn-warning"
            onClick={() => onDecrease(col.id)}
          >
            -
          </button>
          <button
            className="btn btn-primary"
            onClick={() => onIncrease(col.id)}
          >
            +
          </button>
          <button
            className="btn btn-danger"
            onClick={() => onDelete(col.id)}
          >
            Remove
          </button>
        </div>
      </td>
    </tr>
  );
};

export default OrderTableItem;
