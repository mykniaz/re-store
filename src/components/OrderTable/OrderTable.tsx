import * as React from 'react';

import OrderTableItem from './OrderTableItem';

export interface IOrderTableItem {
  id: number;
  title: string;
  count: number;
  total: number;
}

interface IProps {
  items: IOrderTableItem[];
  onIncrease: (id: number) => void;
  onDecrease: (id: number) => void;
  onDelete: (id: number) => void;
}

export const OrderTable: React.SFC<IProps> = ({ items, onIncrease, onDecrease, onDelete }) => {
  const tableRows = items.map((col, colIndex: number) => {
    return (
      <OrderTableItem
        key={col.id}
        col={col}
        colIndex={colIndex}
        onIncrease={onIncrease}
        onDecrease={onDecrease}
        onDelete={onDelete}
      />
    );
  });

  return (
    <table className="table order-table">
      <thead>
        <tr>
          <th scope="col">#</th>
          <th scope="col">Item</th>
          <th scope="col">Count</th>
          <th scope="col">Price</th>
          <th scope="col">Action</th>
        </tr>
      </thead>
      <tbody>
        {tableRows}
      </tbody>
    </table>
  );
};

export default OrderTable;
