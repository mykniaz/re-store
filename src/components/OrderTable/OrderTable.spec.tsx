import * as React from 'react';
import { IOrderTableItem, OrderTable } from './OrderTable';
import reactTestRenderer from 'react-test-renderer';
import { log } from 'util';

const getTestItem = (id: number) => ({
  id,
  title: 'Test Title',
  count: 23,
  total: 332,
});

const getOrderTableElement = (items: IOrderTableItem[]) => {
  return (
    <OrderTable
      items={items}
      onIncrease={() => {}}
      onDecrease={() => {}}
      onDelete={() => {}}
    />
  );
};

test('Orders render test', () => {
  const items = [
    getTestItem(1),
    getTestItem(2),
    getTestItem(3),
    getTestItem(4),
    getTestItem(5),
  ];

  const component = reactTestRenderer.create(getOrderTableElement(items));

  const tree = component.toJSON();

  expect(tree).toMatchSnapshot();
});
