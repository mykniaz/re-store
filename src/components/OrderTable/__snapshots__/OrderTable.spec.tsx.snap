// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Orders render test 1`] = `
<table
  className="table order-table"
>
  <thead>
    <tr>
      <th
        scope="col"
      >
        #
      </th>
      <th
        scope="col"
      >
        Item
      </th>
      <th
        scope="col"
      >
        Count
      </th>
      <th
        scope="col"
      >
        Price
      </th>
      <th
        scope="col"
      >
        Action
      </th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th
        scope="row"
      >
        1
      </th>
      <td>
        Test Title
      </td>
      <td>
        23
      </td>
      <td>
        332
        $
      </td>
      <td>
        <div
          className="btn-group"
        >
          <button
            className="btn btn-warning"
            onClick={[Function]}
          >
            -
          </button>
          <button
            className="btn btn-primary"
            onClick={[Function]}
          >
            +
          </button>
          <button
            className="btn btn-danger"
            onClick={[Function]}
          >
            Remove
          </button>
        </div>
      </td>
    </tr>
    <tr>
      <th
        scope="row"
      >
        2
      </th>
      <td>
        Test Title
      </td>
      <td>
        23
      </td>
      <td>
        332
        $
      </td>
      <td>
        <div
          className="btn-group"
        >
          <button
            className="btn btn-warning"
            onClick={[Function]}
          >
            -
          </button>
          <button
            className="btn btn-primary"
            onClick={[Function]}
          >
            +
          </button>
          <button
            className="btn btn-danger"
            onClick={[Function]}
          >
            Remove
          </button>
        </div>
      </td>
    </tr>
    <tr>
      <th
        scope="row"
      >
        3
      </th>
      <td>
        Test Title
      </td>
      <td>
        23
      </td>
      <td>
        332
        $
      </td>
      <td>
        <div
          className="btn-group"
        >
          <button
            className="btn btn-warning"
            onClick={[Function]}
          >
            -
          </button>
          <button
            className="btn btn-primary"
            onClick={[Function]}
          >
            +
          </button>
          <button
            className="btn btn-danger"
            onClick={[Function]}
          >
            Remove
          </button>
        </div>
      </td>
    </tr>
    <tr>
      <th
        scope="row"
      >
        4
      </th>
      <td>
        Test Title
      </td>
      <td>
        23
      </td>
      <td>
        332
        $
      </td>
      <td>
        <div
          className="btn-group"
        >
          <button
            className="btn btn-warning"
            onClick={[Function]}
          >
            -
          </button>
          <button
            className="btn btn-primary"
            onClick={[Function]}
          >
            +
          </button>
          <button
            className="btn btn-danger"
            onClick={[Function]}
          >
            Remove
          </button>
        </div>
      </td>
    </tr>
    <tr>
      <th
        scope="row"
      >
        5
      </th>
      <td>
        Test Title
      </td>
      <td>
        23
      </td>
      <td>
        332
        $
      </td>
      <td>
        <div
          className="btn-group"
        >
          <button
            className="btn btn-warning"
            onClick={[Function]}
          >
            -
          </button>
          <button
            className="btn btn-primary"
            onClick={[Function]}
          >
            +
          </button>
          <button
            className="btn btn-danger"
            onClick={[Function]}
          >
            Remove
          </button>
        </div>
      </td>
    </tr>
  </tbody>
</table>
`;

exports[`Orders render test 2`] = `
<table
  className="table order-table"
>
  <thead>
    <tr>
      <th
        scope="col"
      >
        #
      </th>
      <th
        scope="col"
      >
        Item
      </th>
      <th
        scope="col"
      >
        Count
      </th>
      <th
        scope="col"
      >
        Price
      </th>
      <th
        scope="col"
      >
        Action
      </th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th
        scope="row"
      >
        1
      </th>
      <td>
        Test Title
      </td>
      <td>
        23
      </td>
      <td>
        332
        $
      </td>
      <td>
        <div
          className="btn-group"
        >
          <button
            className="btn btn-warning"
            onClick={[Function]}
          >
            -
          </button>
          <button
            className="btn btn-primary"
            onClick={[Function]}
          >
            +
          </button>
          <button
            className="btn btn-danger"
            onClick={[Function]}
          >
            Remove
          </button>
        </div>
      </td>
    </tr>
    <tr>
      <th
        scope="row"
      >
        2
      </th>
      <td>
        Test Title
      </td>
      <td>
        23
      </td>
      <td>
        332
        $
      </td>
      <td>
        <div
          className="btn-group"
        >
          <button
            className="btn btn-warning"
            onClick={[Function]}
          >
            -
          </button>
          <button
            className="btn btn-primary"
            onClick={[Function]}
          >
            +
          </button>
          <button
            className="btn btn-danger"
            onClick={[Function]}
          >
            Remove
          </button>
        </div>
      </td>
    </tr>
    <tr>
      <th
        scope="row"
      >
        3
      </th>
      <td>
        Test Title
      </td>
      <td>
        23
      </td>
      <td>
        332
        $
      </td>
      <td>
        <div
          className="btn-group"
        >
          <button
            className="btn btn-warning"
            onClick={[Function]}
          >
            -
          </button>
          <button
            className="btn btn-primary"
            onClick={[Function]}
          >
            +
          </button>
          <button
            className="btn btn-danger"
            onClick={[Function]}
          >
            Remove
          </button>
        </div>
      </td>
    </tr>
    <tr>
      <th
        scope="row"
      >
        4
      </th>
      <td>
        Test Title
      </td>
      <td>
        23
      </td>
      <td>
        332
        $
      </td>
      <td>
        <div
          className="btn-group"
        >
          <button
            className="btn btn-warning"
            onClick={[Function]}
          >
            -
          </button>
          <button
            className="btn btn-primary"
            onClick={[Function]}
          >
            +
          </button>
          <button
            className="btn btn-danger"
            onClick={[Function]}
          >
            Remove
          </button>
        </div>
      </td>
    </tr>
    <tr>
      <th
        scope="row"
      >
        5
      </th>
      <td>
        Test Title
      </td>
      <td>
        23
      </td>
      <td>
        332
        $
      </td>
      <td>
        <div
          className="btn-group"
        >
          <button
            className="btn btn-warning"
            onClick={[Function]}
          >
            -
          </button>
          <button
            className="btn btn-primary"
            onClick={[Function]}
          >
            +
          </button>
          <button
            className="btn btn-danger"
            onClick={[Function]}
          >
            Remove
          </button>
        </div>
      </td>
    </tr>
  </tbody>
</table>
`;
